### PR TITLE
[sdk downgrade] Downgraded sdk usage to normal curl call

### PR DESCRIPTION
**Pull request by mayankamencherla**
- Magento 1.x branch has autoloader issue due to wrong usage of razorpay php sdk
- Removing sdk usage
- Using curl call instead
- Several minor improvements to code: like more documentation and cleaned up some areas

### CODE REVIEW

**Comment by mayankamencherla** [comment]
@ccrims0n This has been tested for the following flows: 
- [x] Normal payment flow
- [x] Redirect based success flow
- [x] Redirect based failure flow

**Comment by mayankamencherla** [comment]
Still to test:
- [x] all try catch blocks
- [x] payment verification exception

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 1]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+
+            if (in_array($httpStatus, $this->successHttpCodes) and isset($responseArray['error']) === false)
+            {
+                return $responseArray;
+            }
+            else
+            {
+                if (!empty($responseArray['error']['code']))
+                {
+                    $error = $responseArray['error']['code'].": ".$responseArray['error']['description'];
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: Single quote.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 2]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+                }
+                else
+                {
+                    $error = "RAZORPAY_ERROR: Invalid Response <br/>".$response;
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: Single quotes

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 3]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
         return Mage::getUrl('razorpay/checkout/success');
     }
 
+    public function getRelativeUrl($name, $data = null)
+    {
+        if ($data)
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: add explicit null check.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 4]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+        if ($response === false)
+        {
+            $error = 'CURL_ERROR: ' . $curlError;
+            Mage::throwException($error);
+        }
+        else
+        {
+            $responseArray = json_decode($response, true);
+
+            if (in_array($httpStatus, $this->successHttpCodes) and isset($responseArray['error']) === false)
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: add proper brackets.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 5]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+            {
+                return $responseArray;
+            }
+            else
+            {
+                if (!empty($responseArray['error']['code']))
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: add explicit check.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 6]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+
+        $method = strtoupper($method);
+
+        //cURL Request
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_USERPWD, $keyId . ":" . $keySecret);
+        curl_setopt($ch, CURLOPT_TIMEOUT, self::REQUEST_TIMEOUT);
+
+        if (is_array($content))
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: add explicit check.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 7]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+        {
+            $data = http_build_query($content);
+        }
+        else if (is_string($content))
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: explicit check.

File: app/code/community/Razorpay/Payments/Helper/Data.php [hunk 8]
```diff
@@ -166,6 +176,102 @@ public function getCallbackUrl()
...
+        {
+            $data = $content;
+        }
+        switch ($method)
```
Comment (ccrims0n) [inline app/code/community/Razorpay/Payments/Helper/Data.php]: newline.

File: app/code/community/Razorpay/Payments/Helper/Data.php (hunk 3 above)
Comment (mayankamencherla) [reply app/code/community/Razorpay/Payments/Helper/Data.php]: ```if (empty($data) === false)```

**Pull request merged**